packages/frontend-v2/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,292 @@
+        return courses.concat(requiredCourses);
+      },
+      []
+    );
+    // console.info("Making", requirements.length, "requests")
+
+    const promises: Promise<ScheduleCourse2<null> | null>[] = [];
+
+    for (const requirement of requirements) {
+      promises.push(
+        // new Promise((res, rej) => {
+        //   res({
+        //     name: "FAKE API CALL, REPLACE!",
+        //     classId: requirement.classId.toString(),
+        //     subject: requirement.subject,
+        //     numCreditsMin: 4,
+        //     numCreditsMax: 4,
+        //     id: null,
+        //   });
+        // })
+        SearchAPI.fetchCourse(
+          requirement.subject,
+          requirement.classId.toString()
+        )
+      );
+    }
+
+    Promise.all(promises).then((courses) => {
+      const courseMap: { [id: string]: ScheduleCourse2<null> } = {};
+      if (courses) {
+        for (const course of courses) {
+          if (course) {
+            courseMap[`${course.subject}${course.classId}`] = course;
+          }
+        }
+        setCourseData(courseMap);
+      }
+    });
+  }, [major.requirementSections]);
+
+  return (
+    <Box p="xs 0px" backgroundColor="neutral.main">
+      <Text
+        py="lg"
+        px="sm"
+        fontSize="xl"
+        color="primary.red.main"
+        fontWeight={700}
+      >
+        {major.name}
+      </Text>
+      {courseData &&
+        major.requirementSections.map((section, index) => (
+          <SidebarSection
+            key={section.title}
+            section={section}
+            courseData={courseData}
+            dndIdPrefix={"sidebar-" + index}
+          />
+        ))}
+    </Box>
+  );
+});
+
+// We need to manually set the display name like this because
+// of how we're using memo above.
+Sidebar.displayName = "Sidebar";
+
+export { Sidebar };
+
+const SidebarSection: React.FC<SidebarSectionProps> = ({
+  section,
+  courseData,
+  dndIdPrefix,
+}) => {
+  const [opened, setOpened] = useState(false);
+  return (
+    <Box
+      backgroundColor="neutral.main"
+      borderTop="1px solid white"
+      cursor="pointer"
+      userSelect="none"
+    >
+      <Text
+        onClick={() => {
+          setOpened(!opened);
+        }}
+        color="dark.main"
+        fontWeight="bold"
+        py="md"
+        px="sm"
+      >
+        {section.title}
+      </Text>
+      <Box
+        style={{ display: opened ? "" : "none" }}
+        backgroundColor="neutral.900"
+        padding="10px 20px 15px 10px"
+        cursor="default"
+      >
+        {section.requirements.map((requirement, index) => (
+          <SectionRequirement
+            requirement={requirement}
+            courseData={courseData}
+            dndIdPrefix={dndIdPrefix + "-" + index}
+            key={index}
+          />
+        ))}
+      </Box>
+    </Box>
+  );
+};
+
+const SectionRequirement: React.FC<SidebarRequirementProps> = ({
```

Comment:
I think this + all the cases below could be its own separate file since the logic for it is kind of grouped together, but up to you to decide!

Reply:
Good idea, we refactored the Sidebar file into SidebarSection and SectionRequirement.